### PR TITLE
Rework edit user endpoint

### PR DESCRIPTION
**Pull request by TIove**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/161

### CODE REVIEW

**Review by 24eeeee — APPROVED** [review]

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            if (request == null)
+            {
+                throw new BadRequestException();
+            }
+            
+            var result = new JsonPatchDocument<DbUser>();
+
+            foreach (var item in request.Operations)
+            {
+                if (item.path == $"/{nameof(EditUserRequest.AvatarImage)}")
```
Comment (TIove) [inline src/UserService.Mappers/Models/PatchDbUserMapper.cs:68]: Change case sensetive

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+                if (item.path == $"/{nameof(EditUserRequest.Status)}")
+                {
+                    item.value = (int) item.value;
+                }
+                
+                switch (item.OperationType)
+                {
+                    case OperationType.Add:
+                        result.Operations.Add(new Operation<DbUser>("/add", item.path, item.from, item.value));
```
Comment (TIove) [inline src/UserService.Mappers/Models/PatchDbUserMapper.cs:82]: "/add", "/remove", "/replace" set the var for this

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        private static List<string> Paths => new List<string>
+        {
+            FirstName,
+            LastName,
+            MiddleName,
+            AvatarImage,
+            Status
+        };
+        
+        public static string FirstName => $"/{nameof(EditUserRequest.FirstName)}";
```
Comment (TIove) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:25]: Edit

**Review by MariyaSin — APPROVED** [review]

File: test/UserService.Validation.UnitTests/EditUserRequestValidatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
...IWXMAABYlAAAWJQFJUiTwAAAB02lUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iWE1QIENvcmUgNi4wLjAiPgogICA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPgogICAgICA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIgogICAgICAgICAgICB4bWxuczpleGlmPSJodHRwOi8vbnMuYWRvYmUuY29tL2V4aWYvMS4wLyI+CiAgICAgICAgIDxleGlmOlBpeGVsWURpbWVuc2lvbj4xODwvZXhpZjpQaXhlbFlEaW1lbnNpb24+CiAgICAgICAgIDxleGlmOlBpeGVsWERpbWVuc2lvbj40PC9leGlmOlBpeGVsWERpbWVuc2lvbj4KICAgICAgICAgPGV4aWY6VXNlckNvbW1lbnQ+U2NyZWVuc2hvdDwvZXhpZjpVc2VyQ29tbWVudD4KICAgICAgPC9yZGY6RGVzY3JpcHRpb24+CiAgIDwvcmRmOlJERj4KPC94OnhtcG1ldGE+Croe+NIAAAAcaURPVAAAAAIAAAAAAAAACQAAACgAAAAJAAAACQAAANkEB42SAAAApUlEQVQoFQCZAGb/AXhsa/8lJyYACgYFAPn8+wABXEE//xUPCQDn4+EAKCosAAGQeHb/7OfkAOPh4gAGCwoAAZJ2df/d2toAAAMEABEXFwABd1Zb/+jn5gAgKSgABQsLAAGSbHb/v8DAADhEQAANExIAAaB7gf+xsLUAHyglAB0dGwABe1da//Hw9wAIDQwA//b1AAFsSEv/FhkeAA0JCQD07e0AAAAA//8KREfTAAAAoklEQVQBmQBm/wFSMjb/HB8gAAwFBgAIAAEAAUksMf8eIyMAJRsaAP309QABOyEn/1VcWgBJPjwAzcDAAAE+ICT/QEVDADEoIwDCurkAAVc0OP/2+vYA6+XfAAoGBgABZklP/xIPDgDp490AAAIBAAF9Z3D/APf4APr08QD8Af8AAZqBif/e19kAFREQAAQIBQABjW5z/+7u8gAlIiEA3d7eAK1yazsTrFtEAAAAAElFTkSuQmCC"),
```
Comment (djino97) [inline test/UserService.Validation.UnitTests/EditUserRequestValidatorTests.cs:62]: format string, not more then 120 chars in line

Comment (TIove) [reply test/UserService.Validation.UnitTests/EditUserRequestValidatorTests.cs:62]: I asked Spartak about it on SCRUM, and he said it is ok

**Review by borovHokage — APPROVED** [review]

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        private Guid? GetAvatarImageId(string avatarImage)
+        {
+            Guid? avatarImageId = null;
+            if (!string.IsNullOrEmpty(avatarImage))
+            {
+                try
+                {
+                    Response<IOperationResult<Guid>> response = _requestClient.GetResponse<IOperationResult<Guid>>(
+                        IAddImageRequest.CreateObj(avatarImage),
+                        timeout: RequestTimeout.After(ms: 500)).Result;
```
Comment (BelehovEgor) [inline src/UserService.Mappers/Models/PatchDbUserMapper.cs:29]: Why is the timeout exactly 500 ms, was it on the task or not?

**Review by BelehovEgor — APPROVED** [review]

File: test/UserService.Validation.UnitTests/EditUserRequestValidatorTests.cs (hunk 4 above)
Comment (djino97) [reply test/UserService.Validation.UnitTests/EditUserRequestValidatorTests.cs:62]: Сhanges will be?

**Pull request merged**